configurationproperties-annotationprocessor/src/main/java/com/vaadin/flow/spring/ConfigurationPropertiesAnnotationProcessor.java

```diff
@@ -300,38 +283,62 @@ private void writePostConstruct(PrintWriter out,
                                     List<FieldDescriptor> fields) {
         out.println(" @PostConstruct");
         out.println(" public void transferValuesToSystem() {");
-        fields.forEach(f -> {
-            out.println("  if (" + f.fieldName + " != null) " +
-                    "System.setProperty(\"vaadin." + f.formerPropertyName
-                    + "\", " + f.fieldName + ".toString());");
+        fields.forEach(field -> {
+            String code = "  if (" + field.fieldName + " != null) " +
+                    "System.setProperty(\"vaadin." + field.formerPropertyName
```

Comment:
`System.setProperty` : what does this code do ?
Why does it set System property ?

Do you set every property from `application.properties` as a system ?
What's the reason for this ?

Reply:
I think my answer has disappeared, here (and it was a long one). I will try to rewrite it:

As far as I have seen the vaadin servlet reads the initial parameters from the `DeploymentConfiguration` class. It reads the values either from web.xml or from the system properties. Hence why I set them as system properties.

Also, please note that the expected name for each property can be different than the property name set by the user, due to the relaxing naming scheme from springboot. Hence why I do this, to warranty that property names are set as expected.

When I faced this point, I also managed the option of serializing this object to a property map, so it could be passed to `ServletRegistrationBean::setInitparameters` inside `SpringBootAutoConfiguration::servletRegistrationBean`. That would avoid setting system properties, though I discarded this option as I thought it would be less intrusive the other way.